migrations/2018_09_29_060444_replace-emoji-shorcutes-with-unicode.php

```diff
@@ -0,0 +1,109 @@
+        return chr(($num >> 6) + 192).chr(($num & 63) + 128);
+    }
+
+    if ($num <= 0xFFFF) {
+        return chr(($num >> 12) + 224).chr((($num >> 6) & 63) + 128).chr(($num & 63) + 128);
+    }
+
+    return chr(($num >> 18) + 240).chr((($num >> 12) & 63) + 128).chr((($num >> 6) & 63) + 128).chr(($num & 63) + 128);
+}
+
+function cleanupCode($code)
+{
+    $currectionMap = [
+        '1f468-2764-1f468' => '1f468-200d-2764-fe0f-200d-1f468',
+        '1f469-2764-1f469' => '1f469-200d-2764-fe0f-200d-1f469',
+        '1f468-2764-1f48b-1f468' => '1f468-200d-2764-fe0f-200d-1f48b-200d-1f468',
+        '1f469-2764-1f48b-1f469' => '1f469-200d-2764-fe0f-200d-1f48b-200d-1f469',
+        '1f468-1f468-1f466' => '1f468-200d-1f468-200d-1f466',
+        '1f468-1f468-1f466-1f466' => '1f468-200d-1f468-200d-1f466-200d-1f466',
+        '1f468-1f468-1f467' => '1f468-200d-1f468-200d-1f467',
+        '1f468-1f468-1f467-1f466' => '1f468-200d-1f468-200d-1f467-200d-1f466',
+        '1f468-1f468-1f467-1f467' => '1f468-200d-1f468-200d-1f467-200d-1f467',
+        '1f468-1f469-1f466-1f466' => '1f468-200d-1f469-200d-1f466-200d-1f466',
+        '1f468-1f469-1f467' => '1f468-200d-1f469-200d-1f467',
+        '1f468-1f469-1f467-1f466' => '1f468-200d-1f469-200d-1f467-200d-1f466',
+        '1f468-1f469-1f467-1f467' => '1f468-200d-1f469-200d-1f467-200d-1f467',
+        '1f469-1f469-1f466' => '1f469-200d-1f469-200d-1f466',
+        '1f469-1f469-1f466-1f466' => '1f469-200d-1f469-200d-1f466-200d-1f466',
+        '1f469-1f469-1f467' => '1f469-200d-1f469-200d-1f467',
+        '1f469-1f469-1f467-1f466' => '1f469-200d-1f469-200d-1f467-200d-1f466',
+        '1f469-1f469-1f467-1f467' => '1f469-200d-1f469-200d-1f467-200d-1f467',
+        '1f441-1f5e8' => '1f441-200d-1f5e8', // as always PITA
+        '1f3f3-1f308' => '1f3f3-fe0f-200d-1f308',
+
+        // https://github.com/twitter/twemoji/issues/192
+        '1f91d-1f3fb' => '1f91d',
+        '1f91d-1f3fc' => '1f91d',
+        '1f91d-1f3fd' => '1f91d',
+        '1f91d-1f3fe' => '1f91d',
+        '1f91d-1f3ff' => '1f91d',
+        '1f93c-1f3fb' => '1f93c',
+        '1f93c-1f3fc' => '1f93c',
+        '1f93c-1f3fd' => '1f93c',
+        '1f93c-1f3fe' => '1f93c',
+        '1f93c-1f3ff' => '1f93c',
+    ];
+
+    if (isset($currectionMap[$code])) {
+        return $currectionMap[$code];
+    }
+
+    return  ltrim($code, '0');
+}
+
+function convertToEmoji($code)
+{
+    $codepints = explode('-', $code);
+
+    return implode('', array_map('fromCodePoint', $codepints));
+}
+
+return [
+    'up' => function (Builder $schema) {
+        $posts = $schema->getConnection()->table('posts')
+            ->select('id', 'content')
+            ->where('content', 'like', '%<EMOJI%')
+            ->cursor();
+
+        foreach ($posts as $post) {
+            $content = preg_replace_callback(
+                '/<EMOJI seq="(.+?)">.+?<\/EMOJI>/',
+                function ($m) {
+                    return convertToEmoji(cleanupCode($m[1]));
```

Comment:
Could you inline these functions, to avoid having side effects (the definition of functions) in this file?

The `$currectionMap` (typo!) array could be a variable that you `use`, that is fine.

Reply:
`$codepints` is another typo.